### PR TITLE
[#13326] App Check Tests

### LINKED ISSUE #13326: Expand test coverage of App Check
### Description

Some methods (e.g. `getLimitedUseTokenWithCompletion:` are untested.

### API Proposal

_No response_

### Firebase Product(s)

App Check

### PR DESCRIPTION
**Pull request by yakovmanshin**
[Closes #13326]

* Added tests for `getLimitedUseTokenWithCompletion:` in `FIRAppAttestProvider`, `FIRDeviceCheckProvider`, and `FIRAppCheckDebugProvider`
* Added a test for `FIRAppCheckDebugProvider`’s `localDebugToken`
* Introduced the `FIRAppCheckValidatorTests` suite
* Marked `FIRAppCheckValidator`’s `init` as unavailable

### CODE REVIEW

File: FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckValidatorTests.m [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+
+#import "FirebaseAppCheck/Sources/Core/FIRAppCheckValidator.h"
+#import "FirebaseCore/FIROptions.h"
```
Comment (paulb777) [inline FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckValidatorTests.m:20]: Unit tests should use `#import "FirebaseCore/Extension/FIROptionsInternal.h"`

**Review by paulb777 — COMMENTED** [review]
Thanks for the improving the AppCheck test coverage!

The CI issue can be tested locally by running `scripts/check_imports.swift`

File: FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckValidatorTests.m (hunk 1 above)
Comment (yakovmanshin) [reply FirebaseAppCheck/Tests/Unit/Core/FIRAppCheckValidatorTests.m]: Fixed ✅

`check_imports` should pass now. Hopefully it was the only failing check.

**Review by ncooke3 — APPROVED** [review]
Thanks, @yakovmanshin!

**Review by paulb777 — APPROVED** [review]

**Pull request merged**